buildenv/jenkins/common/variables-functions.groovy

```diff
@@ -637,14 +637,38 @@ def get_date() {
 }
 
 /*
-* Set TESTS_TARGETS, indicating the level of testing
-* and sets TEST_FLAG for all targets if defined in variable file
+* Set TESTS_TARGETS, indicating the level of testing.
 */
 def set_test_targets() {
-    TESTS_TARGETS = params.TESTS_TARGETS
-    if (!TESTS_TARGETS) {
-        // set default TESTS_TARGETS for pipeline job (fetch from variables file)
-        TESTS_TARGETS = get_default_test_targets()
+    TESTS_TARGETS = (params.TESTS_TARGETS) ? params.TESTS_TARGETS.trim() : "none"
+    TARGET_NAMES = []
+
+    if (TESTS_TARGETS != 'none') {
+        for (target in TESTS_TARGETS.replaceAll("\\s","").toLowerCase().tokenize(',')) {
+            switch (target) {
+                case "sanity":
+                case "extended":
+                    TARGET_NAMES.add("${target}.functional")
+                    break
+                default:
+                    TARGET_NAMES.add(target)
+            }
+        }
+    }
+}
+
+/*
+* Set TEST_FLAG for all targets if defined in variable file.
+* Set EXCLUDED_TESTS if defined in variable file.
+* Set EXTRA_TEST_LABELS map if defined in variable file.
+*/
+def set_test_misc() {
+    EXTRA_TEST_LABELS = [:]
```

Comment:
Extra test labels can be added only in the variables file, no build parameters available. Should users be able to set them in the UI when a build is launched?

Reply:
I don't think it's a necessary use case. Can be added later if needed. Use in a grinder would be more likely.